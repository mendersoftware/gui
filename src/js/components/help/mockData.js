// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {
    hasDeviceConfig: true,
    hasMonitor: true
  },
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '2003-01-01'
  }
};
