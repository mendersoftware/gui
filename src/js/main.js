import React from 'react';
import { Router, Route, hashHistory } from 'react-router';
import { render } from 'react-dom';
import { MuiThemeProvider } from 'material-ui/styles';
var routes = require('./config/routes');


import injectTapEventPlugin from 'react-tap-event-plugin';

//Needed for onTouchTap
//Can go away when react 1.0 release
//Check this repo:
//https://github.com/zilverline/react-tap-event-plugin
injectTapEventPlugin();

render((
  <MuiThemeProvider>
    <Router history={hashHistory}>
      {routes}
    </Router>
  </MuiThemeProvider>
), document.getElementById('main'))
