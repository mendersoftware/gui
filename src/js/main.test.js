// Copyright 2020 Northern.tech AS
//
//    Licensed under the Apache License, Version 2.0 (the "License");
//    you may not use this file except in compliance with the License.
//    You may obtain a copy of the License at
//
//        http://www.apache.org/licenses/LICENSE-2.0
//
//    Unless required by applicable law or agreed to in writing, software
//    distributed under the License is distributed on an "AS IS" BASIS,
//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//    See the License for the specific language governing permissions and
//    limitations under the License.
import React from 'react';
import Linkify from 'react-linkify';
import * as router from 'react-router-dom';

import { prettyDOM } from '@testing-library/dom';
import { render, screen, waitFor } from '@testing-library/react';
import 'jsdom-worker';

import { mockDate, undefineds } from '../../tests/mockData';
import GeneralApi from './api/general-api';
import { AppProviders } from './main';

jest.mock('react-linkify');

describe('Main Component', () => {
  it('renders correctly', async () => {
    Linkify.default = jest.fn();
    Linkify.default.mockReturnValue(null);
    const { MemoryRouter } = router;
    window.localStorage.getItem.mockReturnValueOnce('false');
    const MockBrowserRouter = ({ children }) => <MemoryRouter initialEntries={['/ui']}>{children}</MemoryRouter>;
    // eslint-disable-next-line
    router.BrowserRouter = MockBrowserRouter;
    const ui = <AppProviders />;
    const post = jest.spyOn(GeneralApi, 'post');
    jest.setSystemTime(mockDate);
    const { baseElement, rerender } = render(ui);
    await waitFor(() => screen.queryByText('Software distribution'), { timeout: 2000 });
    await waitFor(() => rerender(ui));
    const view = prettyDOM(baseElement.firstChild, 100000, { highlight: false })
      .replace(/id="mui-[0-9]*"/g, '')
      .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, '')
      .replace(/\\/g, '');
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
    let settingsCall;
    const keys = ['approach', 'artifactIncluded', 'complete', 'deviceType', 'progress'];
    await waitFor(
      () => {
        settingsCall = post.mock.calls.filter(
          ([target, body]) => target === '/api/management/v1/useradm/settings/me' && keys.every(key => Object.keys(body.onboarding).includes(key))
        );
        return expect(settingsCall.length).toBeTruthy();
      },
      { timeout: 5000 }
    );
    expect(settingsCall).toBeTruthy();
  }, 10000);
});
