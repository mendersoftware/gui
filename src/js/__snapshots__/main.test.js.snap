// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component renders correctly 1`] = `
"<div>
  <div
    id="app"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular mui-18fwrck-MuiToolbar-root-header"
      id="fixedHeader"
    >
      <div
        class="flexbox centered fadeInSlow mui-2rm9gj-demoTrialAnnouncement"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall mui-e9dld4-MuiSvgIcon-root-redAnnouncementIcon"
          data-testid="AnnouncementIcon"
          focusable="false"
          style="margin-right: 4px; min-width: 24px;"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 2H4c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 9h-2V5h2v6zm0 4h-2v-2h2v2z"
          />
        </svg>
        <p
          style="margin: 4px;"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-tsicd0-MuiSvgIcon-root-demoAnnouncementIcon"
          data-testid="CloseIcon"
          focusable="false"
          style="margin-left: 4px; cursor: pointer;"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
      <div
        class="flexbox space-between"
      >
        <div
          class="flexbox center-aligned"
        >
          <a
            href="/"
          >
            <img
              id="logo"
              src="headerlogo.png"
            />
          </a>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root mui-j8sktw-MuiFormControl-root-MuiTextField-root-root"
        >
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart mui-s15sbk-MuiInputBase-root-MuiInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-standard MuiInputAdornment-sizeSmall mui-ittuaa-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeSize mui-8zz2bg-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart mui-nqgi9r-MuiInputBase-input-MuiInput-input"
              
              placeholder="Search devices"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="flexbox center-aligned"
        >
          <div
            class="header-section"
            data-mui-internal-clone-element="true"
          >
            <a
              class=""
              href="/devices"
            >
              <span>
                2
              </span>
              <span
                id="limit"
              >
                /
                500
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-1lqc678-MuiSvgIcon-root"
                data-testid="DeveloperBoardIcon"
                focusable="false"
                style="margin: 0px 7px 0px 10px; font-size: 20px;"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 9V7h-2V5c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2v-2h2v-2h-2v-2h2v-2h-2V9h2zm-4 10H4V5h14v14zM6 13h5v4H6zm6-6h4v3h-4zM6 7h5v5H6zm6 4h4v6h-4z"
                />
              </svg>
            </a>
            <a
              href="/devices/pending"
              style="margin-left: 7px;"
            >
              1
               pending
            </a>
          </div>
          <a
            class="header-section"
            href="/deployments/active"
          >
            <span>
              2
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium flip-horizontal mui-18cdzti-MuiSvgIcon-root-icon"
              data-testid="RefreshIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
              />
            </svg>
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium header-dropdown mui-12pkhju-MuiButtonBase-root-MuiButton-root-dropDown"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium mui-gcc2o7-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-dax6cs-MuiSvgIcon-root-buttonColor"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88C7.55 15.8 9.68 15 12 15s4.45.8 6.14 2.12C16.43 19.18 14.03 20 12 20z"
                />
              </svg>
            </span>
            a2@b.com
            <span
              class="MuiButton-endIcon MuiButton-iconSizeMedium mui-1gnd1fd-MuiButton-endIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-1lqc678-MuiSvgIcon-root"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m7 10 5 5 5-5z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root mui-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="leftFixed leftNav mui-vketa7-list"
    >
      <ul
        class="MuiList-root MuiList-padding mui-h4y409-MuiList-root"
        style="padding: 0px;"
      >
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav mui-s9istv-MuiListItem-root-navLink"
          href="/"
        >
          <div
            class="MuiListItemText-root mui-924vb1-MuiListItemText-root"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary mui-mon4qq-MuiTypography-root"
            >
              Dashboard
            </span>
          </div>
        </a>
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav mui-s9istv-MuiListItem-root-navLink"
          href="/devices"
        >
          <div
            class="MuiListItemText-root mui-924vb1-MuiListItemText-root"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary mui-mon4qq-MuiTypography-root"
            >
              Devices
            </span>
          </div>
        </a>
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav mui-s9istv-MuiListItem-root-navLink"
          href="/releases"
        >
          <div
            class="MuiListItemText-root mui-924vb1-MuiListItemText-root"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary mui-mon4qq-MuiTypography-root"
            >
              Releases
            </span>
          </div>
        </a>
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav mui-s9istv-MuiListItem-root-navLink"
          href="/deployments"
        >
          <div
            class="MuiListItemText-root mui-924vb1-MuiListItemText-root"
            style="text-transform: uppercase;"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary mui-mon4qq-MuiTypography-root"
            >
              Deployments
            </span>
          </div>
        </a>
      </ul>
      <ul
        class="MuiList-root MuiList-padding mui-anvol5-MuiList-root-infoList"
      >
        <a
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav mui-16kmypy-MuiListItem-root-listItem"
          href="/help"
        >
          <div
            class="MuiListItemText-root mui-924vb1-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary mui-mon4qq-MuiTypography-root"
            >
              Help & support
            </span>
          </div>
        </a>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding mui-16kmypy-MuiListItem-root-listItem"
        >
          <div
            class="MuiListItemText-root MuiListItemText-multiline mui-1d3wby6-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary mui-mon4qq-MuiTypography-root"
            >
              <div
                class="clickable slightly-smaller"
                data-mui-internal-clone-element="true"
              >
                Version: master
              </div>
            </span>
            <p
              class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary mui-6djxp6-MuiTypography-root"
            >
              <a
                class="mui-9ca4f1-licenseLink"
                href="https://docs.mender.io/development/release-information/open-source-licenses"
                rel="noopener noreferrer"
                target="_blank"
              >
                License information
              </a>
            </p>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="rightFluid container"
    >
      <h4
        class="margin-left-small"
      >
        Dashboard
      </h4>
      <div
        class="mui-1g6rhfz-board"
      >
        <div
          class="mui-1vfij3o-left"
        >
          <div
            class="dashboard"
          >
            <div
              class="widget "
            >
              <div
                class="flexbox widgetHeader"
              >
                <div
                  class="flexbox center-aligned"
                >
                  Accepted devices 
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium margin-left-small green mui-1lqc678-MuiSvgIcon-root"
                    data-testid="CheckCircleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="flexbox column widgetMainContent"
              >
                <div
                  class="counter"
                >
                  2
                </div>
              </div>
            </div>
            <div
              class="widget mui-169u0zx-widget"
            >
              <div
                class="flexbox widgetHeader"
              >
                Devices with issues
              </div>
              <div
                class="widgetMainContent"
              >
                <a
                  class="flexbox center-aligned column mui-1i32ysv-issueType green"
                  href="/devices?issues=offline"
                >
                  <div
                    class="relative mui-1donq9n-issueContainer"
                  >
                    <div
                      class="mui-1uwsjn9-issueCounter"
                    >
                      0
                    </div>
                    <div
                      class="absolute mui-wx993p-issueWrapper"
                    >
                      <div
                        class="relative"
                      >
                        <div
                          class="absolute"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall absolute mui-1cwsc1r-MuiSvgIcon-root"
                          data-testid="WifiOffIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M22.99 9C19.15 5.16 13.8 3.76 8.84 4.78l2.52 2.52c3.47-.17 6.99 1.05 9.63 3.7l2-2zm-4 4c-1.29-1.29-2.84-2.13-4.49-2.56l3.53 3.53.96-.97zM2 3.05 5.07 6.1C3.6 6.82 2.22 7.78 1 9l1.99 2c1.24-1.24 2.67-2.16 4.2-2.77l2.24 2.24C7.81 10.89 6.27 11.73 5 13v.01L6.99 15c1.36-1.36 3.14-2.04 4.92-2.06L18.98 20l1.27-1.26L3.29 1.79 2 3.05zM9 17l3 3 3-3c-1.65-1.66-4.34-1.66-6 0z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <p
                    class="align-center slightly-smaller"
                  >
                    Offline
                  </p>
                </a>
                <a
                  class="flexbox center-aligned column mui-1i32ysv-issueType green"
                  href="/devices?issues=authRequests"
                >
                  <div
                    class="relative mui-1donq9n-issueContainer"
                  >
                    <div
                      class="mui-1uwsjn9-issueCounter"
                    >
                      0
                    </div>
                    <div
                      class="absolute mui-wx993p-issueWrapper"
                    >
                      <div
                        class="relative"
                      >
                        <div
                          class="absolute"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall absolute mui-1cwsc1r-MuiSvgIcon-root"
                          data-testid="VpnKeyOutlinedIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M22 19h-6v-4h-2.68c-1.14 2.42-3.6 4-6.32 4-3.86 0-7-3.14-7-7s3.14-7 7-7c2.72 0 5.17 1.58 6.32 4H24v6h-2v4zm-4-2h2v-4h2v-2H11.94l-.23-.67C11.01 8.34 9.11 7 7 7c-2.76 0-5 2.24-5 5s2.24 5 5 5c2.11 0 4.01-1.34 4.71-3.33l.23-.67H18v4zM7 15c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3zm0-4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <p
                    class="align-center slightly-smaller"
                  >
                    Authentication requests
                  </p>
                </a>
              </div>
            </div>
            <div
              class="widget flexbox centered"
            >
              <p
                class="muted"
                style="max-width: 200px;"
              >
                + connect more devices
              </p>
            </div>
          </div>
          <div
            class="flexbox centered"
          >
            <p
              class="mui-10smx4s-default icon flexbox center-aligned mui-urjt41-default "
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall mui-1cwsc1r-MuiSvgIcon-root"
                data-testid="InfoOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
              <span>
                With a more advanced commercial Mender plan you get access to actionable insights into the devices you are updating with Mender. Get an overview of the Mender plans on the 
                <a
                  href="https://mender.io/plans/features"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  features page
                </a>
                .
              </span>
            </p>
          </div>
        </div>
        <div
          class="mui-1lmf2ic-right deployments"
        >
          <div
            class="dashboard flexbox column"
            style="grid-template-columns: 1fr; row-gap: 10px;"
          >
            <h4
              class="margin-bottom-none margin-left-small"
            >
              Recent deployments
            </h4>
            <h5
              class="margin-bottom-none"
            >
              Pending
            </h5>
            <div
              class="clickable mui-14kqbng-base"
            >
              <div>
                <div>
                  r1
                </div>
                <div>
                  test deployment
                </div>
              </div>
              <div
                class="flexbox center-aligned"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit mui-1kjkxoc-MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2M12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4M12,10.5A1.5,1.5 0 0,1 13.5,12A1.5,1.5 0 0,1 12,13.5A1.5,1.5 0 0,1 10.5,12A1.5,1.5 0 0,1 12,10.5M7.5,10.5A1.5,1.5 0 0,1 9,12A1.5,1.5 0 0,1 7.5,13.5A1.5,1.5 0 0,1 6,12A1.5,1.5 0 0,1 7.5,10.5M16.5,10.5A1.5,1.5 0 0,1 18,12A1.5,1.5 0 0,1 16.5,13.5A1.5,1.5 0 0,1 15,12A1.5,1.5 0 0,1 16.5,10.5Z"
                  />
                </svg>
                <span
                  class="margin-left-small"
                >
                  Queued to start
                </span>
              </div>
            </div>
            <div
              class="clickable mui-14kqbng-base"
            >
              <div>
                <div>
                  r1
                </div>
                <div>
                  test deployment 2
                </div>
              </div>
              <div
                class="flexbox center-aligned"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit mui-1kjkxoc-MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2M12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4M12,10.5A1.5,1.5 0 0,1 13.5,12A1.5,1.5 0 0,1 12,13.5A1.5,1.5 0 0,1 10.5,12A1.5,1.5 0 0,1 12,10.5M7.5,10.5A1.5,1.5 0 0,1 9,12A1.5,1.5 0 0,1 7.5,13.5A1.5,1.5 0 0,1 6,12A1.5,1.5 0 0,1 7.5,10.5M16.5,10.5A1.5,1.5 0 0,1 18,12A1.5,1.5 0 0,1 16.5,13.5A1.5,1.5 0 0,1 15,12A1.5,1.5 0 0,1 16.5,10.5Z"
                  />
                </svg>
                <span
                  class="margin-left-small"
                >
                  Queued to start
                </span>
              </div>
            </div>
            <h5
              class="margin-bottom-none"
            >
              In Progress
            </h5>
            <div
              class="clickable mui-14kqbng-base"
            >
              <div>
                <div>
                  r1
                </div>
                <div>
                  test deployment
                </div>
              </div>
              <div
                class="relative mui-f0yq8v-container minimal "
              >
                <div
                  class="chart-container"
                >
                  <div
                    class="progress-chart relative detailed"
                  >
                    <div
                      class="progress-step"
                      style="left: 0%; width: 100%;"
                    >
                      <div
                        class="progress-bar"
                        style="width: 100%; background-color: rgb(170, 170, 170);"
                      />
                      <div
                        style="display: contents;"
                      >
                        <div
                          class="progress-bar green"
                          style="width: 0%;"
                        />
                        <div
                          class="progress-bar warning"
                          style="left: 0%; width: 0%;"
                        />
                      </div>
                    </div>
                    <div
                      class="progress-step relative flexbox progress-step-total"
                    >
                      <div
                        class="progress-bar"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="clickable mui-14kqbng-base"
            >
              <div>
                <div>
                  r1
                </div>
                <div>
                  test deployment 2
                </div>
              </div>
              <div
                class="flexbox center-aligned"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit mui-1kjkxoc-MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2M12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4M12,10.5A1.5,1.5 0 0,1 13.5,12A1.5,1.5 0 0,1 12,13.5A1.5,1.5 0 0,1 10.5,12A1.5,1.5 0 0,1 12,10.5M7.5,10.5A1.5,1.5 0 0,1 9,12A1.5,1.5 0 0,1 7.5,13.5A1.5,1.5 0 0,1 6,12A1.5,1.5 0 0,1 7.5,10.5M16.5,10.5A1.5,1.5 0 0,1 18,12A1.5,1.5 0 0,1 16.5,13.5A1.5,1.5 0 0,1 15,12A1.5,1.5 0 0,1 16.5,10.5Z"
                  />
                </svg>
                <span
                  class="margin-left-small"
                >
                  Queued to start
                </span>
              </div>
            </div>
            <h5
              class="margin-bottom-none"
            >
              Finished
            </h5>
            <div
              class="mui-g32kg5-wrapper"
            >
              <time
                class="muted slightly-smaller"
                datetime="2019-01-13T13:00:00+00:00"
              >
                2019-01-13 13:00
              </time>
              <div
                class="clickable mui-14kqbng-base"
              >
                <div>
                  <div>
                    r1
                  </div>
                  <div>
                    test deployment
                  </div>
                </div>
                <div
                  class="mui-bkzd8c-finished"
                >
                  <div
                    class="flexbox mui-kxl8gn-resultsStatus "
                  >
                    <div
                      aria-label="Skipped"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="skipped_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                    <div
                      aria-label="Pending"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="pending_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                    <div
                      aria-label="In progress"
                      class="flexbox centered "
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="progress_status.png"
                      />
                      <div
                        class="status"
                      >
                        1
                      </div>
                    </div>
                    <div
                      aria-label="Successful"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="success_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                    <div
                      aria-label="Failed"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="error_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="mui-g32kg5-wrapper"
            >
              <time
                class="muted slightly-smaller"
                datetime="2019-01-13T13:00:00+00:00"
              >
                2019-01-13 13:00
              </time>
              <div
                class="clickable mui-14kqbng-base"
              >
                <div>
                  <div>
                    r1
                  </div>
                  <div>
                    test deployment 2
                  </div>
                </div>
                <div
                  class="mui-bkzd8c-finished"
                >
                  <div
                    class="flexbox mui-kxl8gn-resultsStatus "
                  >
                    <div
                      aria-label="Skipped"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="skipped_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                    <div
                      aria-label="Pending"
                      class="flexbox centered "
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="pending_status.png"
                      />
                      <div
                        class="status"
                      >
                        1
                      </div>
                    </div>
                    <div
                      aria-label="In progress"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="progress_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                    <div
                      aria-label="Successful"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="success_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                    <div
                      aria-label="Failed"
                      class="flexbox centered disabled"
                      data-mui-internal-clone-element="true"
                    >
                      <img
                        src="error_status.png"
                      />
                      <div
                        class="status"
                      >
                        0
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <a
              class="margin-top"
              href="/deployments"
            >
              See all deployments
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="mui-qbmtgw-progressBarContainer"
  />
</div>"
`;
